redhat-appstudio/service-provider-integration-operator
# Cleanup AwaitingToken state tokens if binding is removed

### What does this PR do?
Prevents accumulation of obsolete tokens which are not resolved (i.e. in `AwaitingToken`) state and their parent bingings are removed. Now the tokens are cleaed up after the binding dissapears.

Contains some significant integration tests optimization & refactoring to conform those new rules.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
https://issues.redhat.com/browse/SVPI-125


### How to test this PR?
- use `mshaposh/spio:latest`
- create binding, remember linked token name
- remove binging, make sure token is removed as well
